### DISCUSSION TITLE
"Airgapped" encryption

### DISCUSSION THREAD

**Discussion by numpde** [idom-team/idom#871] [General]
This may not be the canonical way to solve the general problem; but I'm thinking of an idom component that is sent to the front-end together with a RSA public key for. For example, for entering credentials for some third-party API. The user JS part of the component/page does the encryption, and the encrypted message is sent to the server. This would allow to do development without SSL, at least.

My question is: is it safe to assume that, for example, the contents of a `idom.html.textarea` will _not_ be sent to the server if no explicit logic (such as 'onChange' events) is implemented for it?

Thanks ~

**Comment by Archmonger** [comment]
As long as your ASGI webserver is set up to handle encryption, all IDOM traffic will be encrypted as well.

**Comment by rmorshea** [comment]
To address your specific question, no. The client does not send the content of the `<textarea>` to the server unless there is an event handler attached to it. However, if *any* event handler (not just `onChange`) is attached and that event is triggered, the `<textarea>`'s content will be sent to the server.

**Comment by numpde** [comment]
Thank you for the clarifications!
